tests/e2e/test_rosacli_upgrade.go

```diff
@@ -0,0 +1,242 @@
+package e2e
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-online/ocm-common/pkg/aws/aws_client"
+
+	"github.com/openshift/rosa/tests/ci/labels"
+	"github.com/openshift/rosa/tests/utils/common"
+	"github.com/openshift/rosa/tests/utils/config"
+	"github.com/openshift/rosa/tests/utils/exec/rosacli"
+	"github.com/openshift/rosa/tests/utils/profilehandler"
+)
+
+var _ = Describe("Cluster Upgrade testing",
+	labels.Feature.Policy,
+	func() {
+		defer GinkgoRecover()
+
+		var (
+			clusterID                string
+			rosaClient               *rosacli.Client
+			arbitraryPolicyService   rosacli.PolicyService
+			clusterService           rosacli.ClusterService
+			arbitraryPoliciesToClean []string
+			awsClient                *aws_client.AWSClient
+			profile                  *profilehandler.Profile
+		)
+
+		BeforeEach(func() {
+			By("Get the cluster")
+			clusterID = config.GetClusterID()
+			Expect(clusterID).ToNot(Equal(""), "ClusterID is required. Please export CLUSTER_ID")
+
+			By("Init the client")
+			rosaClient = rosacli.NewClient()
+			arbitraryPolicyService = rosaClient.Policy
+			clusterService = rosaClient.Cluster
+
+			By("Load the profile")
+			profile = profilehandler.LoadProfileYamlFileByENV()
+		})
+
+		AfterEach(func() {
+			By("Clean remaining resources")
+			err := rosaClient.CleanResources(clusterID)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Delete arbitrary policies")
+			if len(arbitraryPoliciesToClean) > 0 {
+				for _, policyArn := range arbitraryPoliciesToClean {
+					err = awsClient.DeletePolicy(policyArn)
+					Expect(err).To(BeNil())
+				}
+			}
+		})
+
+		It("to upgrade operator-roles and roles then cluster - [id:73731]", labels.Critical, labels.Runtime.Upgrade, func() {
+
+			By("Check the cluster version and compare with the profile to decide if skip this case")
+			output, err := clusterService.DescribeCluster(clusterID)
+			Expect(err).To(BeNil())
+			clusterDetail, err := clusterService.ReflectClusterDescription(output)
+			Expect(err).To(BeNil())
+			clusterVersion := clusterDetail.OpenshiftVersion
+
+			if profile.Name != "rosa-upgrade-y-stream" || profile.Version != clusterVersion {
```

Comment:
Don't use profile.Name for the version detection, use `profile.Version=y-1`

Reply:
And for upgrade testing, y-stream will be `profile.Version="y-1"`, it will never equals clusterVersion